packages/open-payments/src/client/requests.test.ts

```diff
@@ -114,6 +114,46 @@ describe('requests', (): void => {
       )
     })
 
+    test.each`
+      title                      | queryParams
+      ${'all defined values'}    | ${{ first: 5, cursor: 'id' }}
+      ${'some undefined values'} | ${{ first: 5, cursor: undefined }}
+      ${'all undefined values'}  | ${{ first: undefined, cursor: undefined }}
+    `(
+      'properly sets query params with $title',
+      async ({ queryParams }): Promise<void> => {
+        const cleanedQueryParams = Object.fromEntries(
+          Object.entries(queryParams).filter(([_, v]) => v != null)
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ) as any
+
+        const scope = nock(baseUrl)
+          .matchHeader('Signature', (sig) => sig === undefined)
+          .matchHeader('Signature-Input', (sigInput) => sigInput === undefined)
+          .get('/incoming-payments')
+          .query(cleanedQueryParams)
+          .reply(200)
```

Comment:
So for some reason, including an undefined param in 
`nock.get().query({ first: 5, cursor: undefined })` will always cause an `Nock: No match for request` error, even though, when you inspect the query params via a function:

<img width="1022" alt="Screen Shot 2022-12-13 at 7 13 38 PM" src="https://user-images.githubusercontent.com/15069181/207421217-da376ac9-5e1c-4573-8b1a-b88b02c84256.png">

the parameter isn't there. This isn't great but at least we can assert that the `axios.get` is called with the `cleanedQueryParams`